File: models/meal_plan.py
import json
from datetime import datetime
from database import db

class MealPlan(db.Model):
    __tablename__ = 'meal_plans'
    
    id = db.Column(db.Integer, primary_key=True)
    member_id = db.Column(db.Integer, db.ForeignKey('family_members.id'), nullable=False)
    plan_type = db.Column(db.String(20), nullable=False)  # Daily, Weekly
    plan_json = db.Column(db.Text, nullable=False)  # Stored as serialized JSON string
    created_at = db.Column(db.DateTime, default=datetime.utcnow)
    
    @property
    def parsed_plan(self):
        """Parse plan_json column into python dict/list."""
        try:
            return json.loads(self.plan_json)
        except Exception:
            return {}

    def to_dict(self):
        return {
            'id': self.id,
            'member_id': self.member_id,
            'plan_type': self.plan_type,
            'plan': self.parsed_plan,
            'created_at': self.created_at.strftime('%Y-%m-%d %H:%M:%S')
        }

File: models/mealPlan.js
const { DataTypes, Model } = require('sequelize');
const sequelize = require('../database');

class MealPlan extends Model {
  get parsed_plan() {
    try {
      return this.plan_json ? JSON.parse(this.plan_json) : {};
    } catch (e) {
      return {};
    }
  }

  toDict() {
    const formattedCreatedAt = this.created_at
      ? new Date(this.created_at).toISOString().replace('T', ' ').substring(0, 19)
      : '';
    return {
      id: this.id,
      member_id: this.member_id,
      plan_type: this.plan_type,
      plan: this.parsed_plan,
      created_at: formattedCreatedAt
    };
  }
}

MealPlan.init({
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  member_id: {
    type: DataTypes.INTEGER,
    allowNull: false
  },
  plan_type: {
    type: DataTypes.STRING(20),
    allowNull: false
  },
  plan_json: {
    type: DataTypes.TEXT,
    allowNull: false
  },
  created_at: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW
  }
}, {
  sequelize,
  modelName: 'MealPlan',
  tableName: 'meal_plans',
  timestamps: false
});

module.exports = MealPlan;
